backend/populate_catalog/task.py

```diff
@@ -91,8 +95,10 @@ def handle_dataset(version_metadata_key: str) -> None:
     dataset_catalog = Catalog.from_file(f"{storage_bucket_path}/{dataset_prefix}/{CATALOG_KEY}")
 
     dataset_version_metadata = read_file(f"{storage_bucket_path}/{version_metadata_key}")
-
-    dataset_catalog.add_child(dataset_version_metadata, strategy=GeostoreSTACLayoutStrategy())
+    dataset_catalog.add_child(
+        child=dataset_version_metadata,  # type: ignore[arg-type]
```

Comment:
child here is of type STACObject and so fails the type check of Catalog/Collection. 
Couldn't see an easy way to find exact type in our code.

Reply:
Another way around this is an assertion. I've added one.